Tools/Microsoft.WinGet.Source/Microsoft.WinGet.Source/Common/BaseAzureCommand.cs

```diff
@@ -0,0 +1,194 @@
+// -----------------------------------------------------------------------
+// <copyright file="BaseAzureCommand.cs" company="Microsoft Corporation">
+//     Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Microsoft.WinGet.Source.Common
+{
+    using System;
+    using System.Management.Automation;
+    using System.Net.Http;
+    using Azure.Identity;
+    using Azure.ResourceManager;
+    using Azure.ResourceManager.AppService;
+    using Azure.ResourceManager.Resources;
+
+    /// <summary>
+    /// This is the base class for all commands that need to access Azure; namely,
+    /// to obtain the base URL for the REST API & possibly an authentication token
+    /// if the user wants to perform privileged actions.
+    /// </summary>
+    public class BaseAzureCommand : BaseSourceCommand
+    {
+        private WebSiteResource webSiteResource;
+
+        /// <summary>
+        /// Gets or sets the base URL for the REST API, if the user would rather
+        /// supply one manually than have it automatically resolved through Azure.
+        /// </summary>
+        [Alias("BaseUrl")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = Constants.ManualSet,
+            ValueFromPipelineByPropertyName = true)]
+        public string Url { get; set; }
+
+        /// <summary>
+        /// Gets or sets the ID for the Azure subscription that contains the
+        /// Azure function REST source; this way, we can resolve the base URL
+        /// automatically.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = Constants.SubscriptionIdSet,
+            ValueFromPipelineByPropertyName = true)]
+        public string SubscriptionId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the Azure subscription that contains the
+        /// Azure function REST source; this way, we can resolve the base URL
+        /// automatically.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = Constants.SubscriptionNameSet,
+            ValueFromPipelineByPropertyName = true)]
+        public string SubscriptionName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the Azure function REST source; this
+        /// is used to resolve the base URL automatically, and possibly
+        /// the authentication token.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = Constants.SubscriptionIdSet,
+            ValueFromPipelineByPropertyName = true)]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = Constants.SubscriptionNameSet,
+            ValueFromPipelineByPropertyName = true)]
+        public string FunctionName { get; set; }
```

Comment:
Can we change the name of this? Function Name is only one possible configuration. We can go through a FrontDoor for instance. Can we make this URI?

Reply:
Also - make sure we aren't relying on a function name - we should take a whole base URI

Reply:
Or you could have an option for a URI instead of function name.